website/blog/2022-07-27-getting-started-with-the-dbt-semantic-layer.md

```diff
@@ -0,0 +1,165 @@
+
+    type: sum
+    sql: amount
+
+    timestamp: order_date
+    time_grains: [day, week, month, year]
+
+    dimensions:
+      - customer_status
+      - has_coupon_payment
+      - has_bank_transfer_payment
+      - has_credit_card_payment
+      - has_gift_card_payment
+
+    filters:
+      - field: status
+        operator: '='
+        value: "'completed'"
+```
+
+This metric has now been defined in the dbt metadata and can be seen in the DAG! 
+
+![Untitled](/img/blog/2022-07-27-getting-started-with-the-dbt-semantic-layer/metrics-dag.png)
+
+**Running The Metric Package To Calculate the Metric**
+
+> ❗ In the future, this is where Jim and Pam would directly query the proxy server to obtain the metric data (aka revenue). We will now cover what you can do with metrics today.
+
+In order to ensure that both Jim and Pam are pulling from the same dataset with the same definition, we’ll want to create a model that materializes this metric for them - this ensures that they’ll always get the same answer for ARR.
+
+Creating a new model called `revenue.sql` , we’ll enter the following macro to call the metrics package. In this example, we’re not interested in the specific payment types and only want to see revenue broken up by `week`and `customer_status`. 
+
+```sql
+select * 
+from {{ metrics.metric(
+    metric_name='revenue',
+    grain='week',
+    dimensions=['customer_status']
+) }}
+```
+
+This would return a dataset that looks like this:
+
+| period | customer_status | revenue |
+| --- | --- | --- |
+| 2018-01-01 | Churn Risk | 43 |
+| 2018-01-01 | Churned | 0 |
+| 2018-01-01 | Healthy | 26 |
+| 2018-01-08 | Churn Risk | 27 |
+
+Jim and Pam would then be able to reference the `revenue` column within the newly created dataset and never have to worry about the calculation of revenue ever again[^2]! The world is perfect and [balance has been restored.](https://www.youtube.com/watch?v=d1EnW4kn1kg) 
+
+**In the near Future with dbt Server and dbt Cloud proxy Server**
+
+When dbt Server and the dbt Cloud proxy Server release later this year, the flow of how metrics are consumed will change significantly. Your organization will no longer need to materialize each metric within a model in order to take advantage of the metric definition. Instead, you’ll be able to directly query the proxy server with the metric code provided and have the correct dataset returned to your BI tool of choice. 
+
+Additionally, integration partners will have built out experiences around Metrics using the Metadata API to create unique and creative ways for consumers to obtain metric data while abstracting away complexity. For example, a box that allows the user to select from a list of metrics, time grains, dimensions, and secondary calculation and then have the correct information returned to them regardless of the selection!
+
+This is an exciting future where data can be consumed in the context of how most people think about it - **metrics.**
```

Comment:
Do you think this line adds anything? Might be able to cut

Reply:
Agreed. Cut it out